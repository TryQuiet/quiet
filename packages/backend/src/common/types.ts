import type EventStore from 'orbit-db-eventstore'
import { type ChannelMessage, type PublicChannel } from '@quiet/types'

export interface PublicChannelsRepo {
  db: EventStore<ChannelMessage>
  eventsAttached: boolean
}

export interface DirectMessagesRepo {
  db: EventStore<string>
  eventsAttached: boolean
}

export type ChannelInfoResponse = Record<string, PublicChannel>

export class StorageOptions {
  orbitDbDir?: string
  ipfsDir?: string
  createPaths = true
}

export interface IPublicKey {
  halfKey: string
}

export type IMessageThread = string

export class ConnectionsManagerOptions {
  env: {
    appDataPath?: string
    resourcesPath?: string
  } = {}

  bootstrapMultiaddrs?: string[] = []
  createPaths?: boolean = true
}

export interface ILibp2pStatus {
  address: string
  peerId: string
}
