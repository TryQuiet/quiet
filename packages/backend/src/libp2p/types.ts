export enum Libp2pEvents {
    PEER_CONNECTED = 'peerConnected',
    PEER_DISCONNECTED = 'peerDisconnected'
}
