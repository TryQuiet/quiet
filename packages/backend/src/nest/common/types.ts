import { type EventsType } from '@orbitdb/core'
import { type ChannelMessage, type PublicChannel } from '@quiet/types'

export interface PublicChannelsRepo {
  db: EventsType<ChannelMessage>
  eventsAttached: boolean
}

export type ChannelInfoResponse = Record<string, PublicChannel>

export class StorageOptions {
  orbitDbDir?: string
  ipfsDir?: string
  createPaths = true
}

export interface IPublicKey {
  halfKey: string
}

export type IMessageThread = string

export class ConnectionsManagerOptions {
  env: {
    appDataPath?: string
    resourcesPath?: string
  } = {}

  bootstrapMultiaddrs?: string[] = []
  createPaths?: boolean = true
}

export interface ILibp2pStatus {
  address: string
  peerId: string
}
