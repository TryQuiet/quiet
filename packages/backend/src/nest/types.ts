import { IncomingMessage, ServerResponse, Server } from 'http'
import { DefaultEventsMap } from 'socket.io/dist/typed-events'
import { Server as SocketIO } from 'socket.io'

export class ConnectionsManagerTypes {
    options: Partial<ConnectionsManagerOptions>
    socketIOPort: number
    httpTunnelPort?: number
    torAuthCookie?: string
    torControlPort?: number
    torResourcesPath?: string
    torBinaryPath?: string
}

export class ConnectionsManagerOptions {
    env: {
        appDataPath?: string
        resourcesPath?: string
    } = {}

    bootstrapMultiaddrs?: string[] = []
    createPaths?: boolean = true
}

export class ConfigOptions {
    options: Partial<ConnectionsManagerOptions>
    socketIOPort: number
    httpTunnelPort?: number
    torAuthCookie?: string
    torControlPort?: number
    torResourcesPath?: string
    torBinaryPath?: string
    env: {
        appDataPath?: string
        resourcesPath?: string
    } = {}

    bootstrapMultiaddrs?: string[] = []
    createPaths?: boolean = true
}

export class ServerIoProviderTypes {
    server: Server<typeof IncomingMessage, typeof ServerResponse>
    io: SocketIO<DefaultEventsMap, DefaultEventsMap, DefaultEventsMap, any>
}

export interface GetPorts {
    socksPort: number
    libp2pHiddenService: number
    controlPort: number
    dataServer: number
    httpTunnelPort: number
}
