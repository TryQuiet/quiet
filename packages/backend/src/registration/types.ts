export enum RegistrationEvents {
    ERROR = 'error',
    SPAWN_HS_FOR_REGISTRAR = 'spawnHsForRegistrar',
    NEW_USER = 'newUser',
    SET_CERTIFICATES = 'setCertificates'
}
