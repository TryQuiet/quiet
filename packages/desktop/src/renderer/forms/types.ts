import { RegisterOptions } from 'react-hook-form'

export interface FieldProps {
    label: string
    name: string
    type: string
    placeholder: string
}

export interface FieldData {
    fieldProps: FieldProps
    validation: RegisterOptions
}

export type FieldsProps<FormValues> = {
    [fieldName in keyof FormValues]: FieldData
}
