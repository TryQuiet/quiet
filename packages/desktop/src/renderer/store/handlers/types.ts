import { AnyAction, ActionCreator } from 'redux'
import { ActionFunction0, Action, ReduxCompatibleReducer } from 'redux-actions'

export interface ActionsBasicType {
  [k: string]: ActionCreator<AnyAction>
}

export type ActionsType<actions extends ActionsBasicType> = {
  [k in keyof actions]: ReturnType<actions[k]>
}

export interface ActionsCreatorsBasicType {
  [k: string]: (...args: any[]) => ActionFunction0<Action<any>>
}

export type ActionsCreatorsTypes<actionCreators extends ActionsCreatorsBasicType> = {
  [k in keyof actionCreators]: ReturnType<ReturnType<actionCreators[k]>>
}

export interface StoreBasicType {
  [key: string]: ReduxCompatibleReducer<any, any>
}

export type StoreType<reducers extends StoreBasicType> = {
  [k in keyof reducers]: ReturnType<reducers[k]>
}

export type PayloadType<actions extends ActionsType<ActionsBasicType>> = actions[keyof actions]['payload']
