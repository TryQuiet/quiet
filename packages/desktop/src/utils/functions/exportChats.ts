import { MessagesDailyGroups } from '@quiet/types'
import { dialog, BrowserWindow } from '@electron/remote'
import fs from 'fs'

export const exportChats = async (channelName: string, channelMessages: MessagesDailyGroups) => {
  dialog
    .showSaveDialog(BrowserWindow.getAllWindows()[0], {
      title: 'Save file',
      defaultPath: `${channelName}.txt`,
      buttonLabel: 'Save',

      filters: [
        { name: 'txt', extensions: ['txt'] },
        { name: 'All Files', extensions: ['*'] },
      ],
    })
    .then(({ filePath }) => {
      if (filePath) {
        fs.writeFile(filePath, channelMessagesToText(channelMessages), err => {
          if (err) {
            console.log(err)
          }
        })
      }
    })
}

const channelMessagesToText = (channelMessages: MessagesDailyGroups) => {
  return Object.keys(channelMessages)
    .map(day => {
      return channelMessages[day]
        .map(messages => messages.map(message => `[${message.nickname} ${message.date}]\n${message.message}`))
        .join('\n\n')
    })
    .join('\n\n\n')
}
