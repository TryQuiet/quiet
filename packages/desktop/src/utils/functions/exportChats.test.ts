import { channelMessagesToText } from './exportChats'
import { MessagesDailyGroups } from '@quiet/types'

const messages: MessagesDailyGroups = {
  '2023-11-22': [
    [
      {
        id: '1',
        type: 1,
        message: 'Hello, World!',
        createdAt: 1637547600,
        date: '2023-11-22',
        nickname: 'JohnDoe',
        isRegistered: true,
        isDuplicated: false,
        pubKey: 'public_key',
      },
      {
        id: '2',
        type: 1,
        message: 'How are you?',
        createdAt: 1637551200,
        date: '2023-11-22',
        nickname: 'JaneDoe',
        isRegistered: true,
        isDuplicated: false,
        pubKey: 'public_key_2',
      },
      {
        id: '3',
        type: 1,
        message: 'Hi there!',
        createdAt: 1637554800,
        date: '2023-11-22',
        nickname: 'Alice',
        isRegistered: true,
        isDuplicated: false,
        pubKey: 'public_key_3',
      },
    ],
  ],
}

describe('channelMessagesToText', () => {
  it('should convert channel messages to text format', () => {
    const result = channelMessagesToText(messages)
    expect(result).toMatchSnapshot(`
      "[JohnDoe 2023-11-22]
      Hello, World!

      [JaneDoe 2023-11-22]
      How are you?

      [Alice 2023-11-22]
      Hi there!

      [Bob 2023-11-23]
      Good morning!"
    `)
  })
})
