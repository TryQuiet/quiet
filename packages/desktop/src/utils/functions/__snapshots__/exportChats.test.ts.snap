// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`channelMessagesToText should convert channel messages to text format: 
      "[JohnDoe 2023-11-22]
      Hello, World!

      [JaneDoe 2023-11-22]
      How are you?

      [Alice 2023-11-22]
      Hi there!

      [Bob 2023-11-23]
      Good morning!"
     1`] = `
"[JohnDoe 2023-11-22]
Hello, World!,[JaneDoe 2023-11-22]
How are you?,[Alice 2023-11-22]
Hi there!"
`;

exports[`channelMessagesToText should convert channel messages to text format: 
      "[JohnDoe 2023-11-22]
      Hello, World!

      [JaneDoe 2023-11-22]
      How are you?wrgletjr

      [Alice 2023-11-22]
      Hi there!

      [Bob 2023-11-23]
      Good morning!"
     1`] = `
"[JohnDoe 2023-11-22]
Hello, World!,[JaneDoe 2023-11-22]
How are you?,[Alice 2023-11-22]
Hi there!"
`;

exports[`channelMessagesToText should convert channel messages to text format: 
      "[JohnDoe 2023-11-22]
Hello, World!

[JaneDoe 2023-11-22]
How are you?

[Alice 2023-11-22]
Hi there!


[Bob 2023-11-23]
Good morning!"
     1`] = `
"[JohnDoe 2023-11-22]
Hello, World!,[JaneDoe 2023-11-22]
How are you?,[Alice 2023-11-22]
Hi there!"
`;
